service-deployment/bootstrap/hadoop-service/hadoop-configuration/nodemanager-start-service.sh

```diff
@@ -33,7 +33,26 @@ then
   $HADOOP_YARN_HOME/bin/yarn nodemanager
 else
   echo gpu machine
-  nvidia-smi
+
+  # The loop is designed for the node restart or kubelet restart.
+  # Because when kubelet crushed, all service in the node will be started at the same time.
+  # Usually drivers' startup process is much slower than node-manager.
+  # So set the try times to 10. If after 10 time retris, the nm service still can't find gpu.
```

Comment:
> 10 time retris, the nm service still can't find gpu. [](start = 40, length = 53)

shall we generate a warning in this case?

Reply:
No. If we can't successfully run nvidia-smi, I will assume the node's drivers is error. Hadoop service shouldn‘t handle too much drivers‘ logic